# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1532554084 14400
#      Wed Jul 25 17:28:04 2018 -0400
# Branch stable
# Node ID c952d8bf1a1fb293d96c94d0e5b0b445996ee0d6
# Parent  19eba00c0c7de023b1d12fb8ecb2398d400f15e8
consider any non-zero return from uname to be failure (bug #54235)

* oct-uname.cc (uname::init): Handle all non-zero values returned from
octave_uname_wrapper as failure, not just negative values.

diff --git a/liboctave/system/oct-uname.cc b/liboctave/system/oct-uname.cc
--- a/liboctave/system/oct-uname.cc
+++ b/liboctave/system/oct-uname.cc
@@ -37,17 +37,17 @@ namespace octave
     void
     uname::init (void)
     {
       char *sysname, *nodename, *release, *version, *machine;
 
       err = octave_uname_wrapper (&sysname, &nodename, &release,
                                   &version, &machine);
 
-      if (err < 0)
+      if (err)
         msg = std::strerror (errno);
       else
         {
           m_sysname = sysname;
           m_nodename = nodename;
           m_release = release;
           m_version = version;
           m_machine = machine;
